### PR TITLE
Added Docker and CI/CD

### PR DESCRIPTION
**Pull request by ShootGan**
Hi Tadeusz, I finally achieved a production-ready Bolang Dockerfile. Also, I made CI/CD to automate pushing a new build of Bolang's image on push event on the master branch in your repository. All you have to do is to generate a GitHub token and pass it into GitHub secrets. All environmental variables should be descriptive in code. When you meet some troubles, let me know.  To run Bolang interpreter, you have to type: 

```sh
docker run -it ghcr.io/tadeuszjt/bolang:latest
```

or simply `bolang:latest` in the case when you will build that image on your own or push it to docker hub.

To compile anything without getting into an interpreter, you don't need to pass `-it` flag.

Command to build Bolang in the Docker locally:

```sh
docker build -t bolang .
```

Greetings from Paweł, and my co-worker @KrzysztofZawisla

### CODE REVIEW

**Comment by tadeuszjt** [comment]
Hey Paweł

Cool. I have no experience whatsoever with Docker, can you explain what it is and why it would be a benefit to include this functionality?

**Comment by KrzysztofZawisla** [comment]
So simply, Docker lets you run Bolang anywhere. For example, you are working on Windows, and you don't have the ability to use Bolang because some dependencies can't be compiled under Windows. In that case, you can use Bolang thought the Docker that lying your computer about virtual computer (don't misunderstand that with Virtual Machine where hardware has to be virtualized either). Also, you can move the whole process of building Bolang based applications to be built in Docker image. With it, you can scale them horizontally (for example servers) with Kubernetes by running them into containers.

There's a video that explains that topic in simple way -> https://youtu.be/Gjnup-PuquQ

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Docker
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - v*
+jobs:
+  push:
```
Comment (tadeuszjt) [inline .github/workflows/docker-publish.yml:9]: Hey

I like the dockerfile but not sure what this is for, is it necessary?

Comment (KrzysztofZawisla) [reply .github/workflows/docker-publish.yml:9]: The purpose of that workflow is to automatically publish a new dockerized version of Bolang on every update on master. I'm using the same workflow to publish a new version of dockerized my services in some projects. Screen rel -> https://imgur.com/a/XBoQN1X.

**Comment by grxnola** [comment]
These guys have to pad out their CVs somehow.

**Comment by tadeuszjt** [comment]
> 
Well at least Paweł has contributed something other than cynicism to this project

**Comment by grxnola** [comment]
I contributed a bunch about whitespace, if you don't recall...

**Comment by tadeuszjt** [comment]
>
Then open a PR so i can tell you to stop padding out your CV

**Comment by tadeuszjt** [comment]
Hey @ShootGan 

I never accepted the pull request because I only ended up using the Dockerfile. I have temporarily removed it from the project so if you would like to finish this to become a contributor then please change the pull request to only include the docker file:

FROM debian:bullseye
RUN apt-get update -y 
RUN apt-get install -y cabal-install libgc-dev ghc llvm-9-dev zlib1g-dev g++ 
RUN cabal update
RUN cabal install cabal-install
RUN cabal install happy alex llvm-hs llvm-hs-pure HUnit language-c temporary
ADD . /home
WORKDIR /home
RUN cabal install --installdir=/usr/local/bin
ENTRYPOINT ["bolang"] 

It would only be fair because it is based on the work you did and I found it very useful :)

Regards
Tadeusz

**Comment by ShootGan** [comment]
Hi, I didn't notice your answer. I will change this. 
Regards,
Paweł

**Comment by ShootGan** [comment]
@tadeuszjt  done :D

**Comment by KrzysztofZawisla** [comment]
I'm proud of you @ShootGan 🚀

**Pull request closed without merging**